### DISCUSSION TITLE
Unify express middleware APIs

### DISCUSSION THREAD

**Discussion by sidharthachatterjee** [gatsbyjs/gatsby#30975] [Ideas / Feature Requests]
As noted in https://github.com/gatsbyjs/gatsby/issues/12924, there isn't an equivalent way to add middleware for `serve` like there is for `develop` (developMiddleware)

Adding a new API for this like we proposed earlier doesn't seem like a great idea and would expand the API surface unnecessarily. Instead we ought to unify these for the two in v3

**Comment by bradvogel** [comment]
Checking in on this - would you all be willing to take a PR to expose a new serveMiddleware configuration option? We're using a forked version of gatsby that exposes this option, just so we can get access to the express app and mount custom middleware.
